src/Symfony/Component/Form/FormFactory.php

```diff
@@ -126,7 +126,7 @@ public function createBuilderForProperty($class, $property, $data = null, array
 
         // user options may override guessed options
         if ($typeGuess) {
-            $options = array_merge($typeGuess->getOptions(), $options);
+            $options = array_replace_recursive($typeGuess->getOptions(), $options);
```

Comment:
@xabbuh I've tested your suggested code (https://3v4l.org/badZN) but it loses the issue case in two points now:
```php
if ($typeGuess) {
    $guessedOptions = array_merge($typeGuessGetOptions, $guessedOptions);
}

// user options may override guessed options
$options = array_merge($guessedOptions, $options);
```